src/service_info.rs

```diff
@@ -499,6 +499,15 @@ impl TxtProperties {
     pub fn get_property_val_str(&self, key: &str) -> Option<&str> {
         self.get(key).map(|x| x.val_str())
     }
+
+    /// Returns a map of properties, where the key is the property key.
+    pub fn get_property_map_str(&self) -> HashMap<String, String> {
```

Comment:
Can we consume the `TxtProperties`? Something like:

```rust
/// Returns a map of properties, where the key is the property key.
///
/// **This method consumes [`TxtProperties`].**
pub fn into_hashmap(self) -> HashMap<String, String>
```

One can always obtain an `HasMap<&str, &str>` using the current methods provided by `TxtProperties`

Reply:
There are 2 commits. The 2nd commit already consumes the `TxtProperties`.